### ISSUE TITLE
[Bug] Failure to serialize IENetwork into xml/bin

### ISSUE THREAD

**Issue by CoderHam** [openvinotoolkit/openvino#4798]
##### System information (version)

- OpenVINO => 2021.1
- Operating System / Platform => Ubuntu 18.04 LTS
- Problem classification: Model Serialization

##### Detailed description
When trying to serialize a model created using the ngraph and function APIs, the serialization fails with an IR 10 not implemented error

##### Steps to reproduce
Used Python 3.6.9
```
from openvino.inference_engine import IECore, IENetwork
import ngraph as ng

param_x = ng.parameter(shape=[1, 16], dtype=np.float32, name="INPUT0")
param_y = ng.parameter(shape=[1, 16], dtype=np.float32, name="INPUT1")
result0 = ng.subtract(param_x, param_y, name="OUTPUT0")
result1 = ng.subtract(param_x, param_y, name="OUTPUT1")

function = ng.impl.Function([result0, result1], [param_x, param_y], "AddSub")
ie_network = IENetwork(ng.impl.Function.to_capsule(function))
ie_network.serialize("./serialized.xml", "./serialized.bin")
```

I get the following error:

```
  File "x.py", line 18, in <module>
    ie_network.serialize("./serialized.xml", "./serialized.bin")
  File "ie_api.pyx", line 1579, in openvino.inference_engine.ie_api.IENetwork.serialize
RuntimeError: The serialize for IR v10 is not implemented
```
##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [x] There is reproducer code and related data files: images, videos, models, etc.

**Comment by ilya-lavrenov** [comment]
@CoderHam could you please try the latest master or release/2021/3 branch? These branches contain this functionality while for older releases it's really not implemented

**Comment by CoderHam** [comment]
@ilya-lavrenov Thanks for your quick response. I used the `-DENABLE_PYTHON=ON` cmake flag for the Python build but I did not see a wheel produced. How can I generate the wheel file from the `releases/2021/3` branch?
Are models generated using 2021.3 release backward compatible to be run with 2021.1 and 2021.2?

**Comment by CoderHam** [comment]
Alternatively is there any other way to build models from scratch using the openvino python API in releases 2021.1/2021.2?

**Comment by ilya-lavrenov** [comment]
@CoderHam IE python wheel build is enabled only if `-DENABLE_WHEEL=ON` is passed to cmake. Please, try this option.

**Comment by CoderHam** [comment]
I get the following failure when I try to use the following command. Am I missing something?
```
(cd build && cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_CLDNN=OFF -DENABLE_PYTHON=ON -DENABLE_WHEEL=ON -DPYTHON_EXECUTABLE=`which python3.6` -DPYTHON_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython3.6m.so -DPYTHON_INCLUDE_DIR=/usr/include/python3.6 .. && make --jobs=$(nproc --all))
```

```
make[2]: *** No rule to make target 'inference-engine/ie_bridges/python/wheel/_pyngraph', needed by 'inference-engine/ie_bridges/python/wheel/CMakeFiles/ie_wheel'.  Stop.
CMakeFiles/Makefile2:6402: recipe for target 'inference-engine/ie_bridges/python/wheel/CMakeFiles/ie_wheel.dir/all' failed
make[1]: *** [inference-engine/ie_bridges/python/wheel/CMakeFiles/ie_wheel.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Building CXX object inference-engine/tools/vpu/vpu_compile/CMakeFiles/myriad_compile.dir/__/common/vpu_tools_common.cpp.o
[100%] Building CXX object inference-engine/tools/vpu/vpu_compile/CMakeFiles/myriad_compile.dir/main.cpp.o
[100%] Linking CXX executable ../../../../../bin/intel64/Release/myriad_compile
[100%] Built target myriad_compile
[100%] Linking CXX executable ../../../../../bin/intel64/Release/myriad_perfcheck
[100%] Built target myriad_perfcheck
Makefile:181: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by ilya-lavrenov** [comment]
@CoderHam some fixes are done here https://github.com/openvinotoolkit/openvino/pull/4837. 

Previously, in order to build `ie_wheel` you need to enable ngraph python API as well (using `-DNGRAPH_PYTHON_BUILD_ENABLE=ON`). But I've relaxed this dependency 

Could you please try the latest master?

**Comment by CoderHam** [comment]
@ilya-lavrenov I am still getting the following error with master:
```
-- Primitive cache is disabled
-- libGNA::API INTERFACE_LIBRARY : /root/openvino/inference-engine/temp/gna_02.00.00.1047.1/include
-- libGNA::KERNEL SHARED_LIBRARY : /root/openvino/inference-engine/temp/gna_02.00.00.1047.1/linux/x64/libgna.so
-- Looking for C++ include unistd.h
-- Looking for C++ include unistd.h - found
-- Looking for C++ include stdint.h
-- Looking for C++ include stdint.h - found
-- Looking for C++ include sys/types.h
-- Looking for C++ include sys/types.h - found
-- Looking for C++ include fnmatch.h
-- Looking for C++ include fnmatch.h - found
-- Looking for strtoll
-- Looking for strtoll - found
-- Found Cython: /usr/local/bin/cython  
CMake Error at inference-engine/ie_bridges/python/wheel/CMakeLists.txt:49 (message):
  patchelf is not found, which is needed to build ie_wheel


-- Configuring incomplete, errors occurred!
```

What patchelf is being referred to? Where can I get the same?

**Comment by CoderHam** [comment]
Ignore the previous comment. I needed additional deps that are described [here](https://github.com/openvinotoolkit/openvino/tree/master/inference-engine/ie_bridges/python).

**Comment by CoderHam** [comment]
When I install the wheel file I see the contents as 
```
/usr/local/lib/python3.6/dist-packages/openvino
`-- libs
    |-- libHeteroPlugin.so
    |-- libMKLDNNPlugin.so
    |-- libMultiDevicePlugin.so
    |-- libhwloc.so.5
    |-- libinference_engine.so
    |-- libinference_engine_ir_reader.so
    |-- libinference_engine_legacy.so
    |-- libinference_engine_lp_transformations.so
    |-- libinference_engine_onnx_reader.so
    |-- libinference_engine_preproc.so
    |-- libinference_engine_snippets.so
    |-- libinference_engine_transformations.so
    |-- libmyriadPlugin.so
    |-- libngraph.so
    |-- libonnx_importer.so
    |-- libonnx_proto.so
    |-- libprotobuf.so
    |-- libprotobuf.so.3.7.1.0
    |-- libtbb.so
    |-- libtbb.so.2
    |-- libtbb_preview.so
    |-- libtbb_preview.so.2
    |-- libtbbbind.so
    |-- libtbbbind.so.2
    |-- libtbbmalloc.so
    |-- libtbbmalloc.so.2
    |-- libtbbmalloc_proxy.so
    |-- libtbbmalloc_proxy.so.2
    |-- myriad_compile
    |-- pcie-ma2x8x.mvcmd
    |-- plugins.xml
    `-- usb-ma2x8x.mvcmd
```

And due to the same I get an import error for the inference_engine module. Do I need to configure the build differently to enable this?

```
>>> from openvino.inference_engine import IECore, IENetwork
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'openvino.inference_engine'
```

**Comment by ilya-lavrenov** [comment]
@mryzhov could you please help?

BTW, I don't see python API library itself..

**Comment by CoderHam** [comment]
Post build I find it in this location
`./bin/intel64/Release/lib/wheels/openvino-0.0.0-000-py3-none-manylinux2014_x86_64.whl`
 whereas the unpacked library itself is in 
`./bin/intel64/Release/lib/python_api/python3.6/openvino/`

**Comment by CoderHam** [comment]
@ilya-lavrenov I tried using the unpacked python library but got the following error

```
  import ngraph as ng
  File "/usr/local/lib/python3.6/dist-packages/ngraph/__init__.py", line 25, in <module>
    from ngraph.ops import absolute
  File "/usr/local/lib/python3.6/dist-packages/ngraph/ops.py", line 20, in <module>
    from ngraph.impl import AxisSet, AxisVector, Coordinate, CoordinateDiff, Function, Node, \
  File "/usr/local/lib/python3.6/dist-packages/ngraph/impl/__init__.py", line 35, in <module>
    from _pyngraph import Function
ImportError: /usr/local/lib/python3.6/dist-packages/_pyngraph.cpython-36m-x86_64-linux-gnu.so: undefined symbol: _ZNK6ngraph4Node11descriptionEv
```

**Comment by mryzhov** [comment]
> @mryzhov could you please help?
> 
> BTW, I don't see python API library itself..

The ie_api.so should be located in another directory (/usr/local/lib/python3.6/dist-packages/openvino/inference_engine/)

**Comment by mryzhov** [comment]
> @ilya-lavrenov When I install the wheel file I see the contents as
> 
> ```
> /usr/local/lib/python3.6/dist-packages/openvino
> `-- libs
>     |-- libHeteroPlugin.so
>     |-- libMKLDNNPlugin.so
>     |-- libMultiDevicePlugin.so
>     |-- libhwloc.so.5
>     |-- libinference_engine.so
>     |-- libinference_engine_ir_reader.so
>     |-- libinference_engine_legacy.so
>     |-- libinference_engine_lp_transformations.so
>     |-- libinference_engine_onnx_reader.so
>     |-- libinference_engine_preproc.so
>     |-- libinference_engine_snippets.so
>     |-- libinference_engine_transformations.so
>     |-- libmyriadPlugin.so
>     |-- libngraph.so
>     |-- libonnx_importer.so
>     |-- libonnx_proto.so
>     |-- libprotobuf.so
>     |-- libprotobuf.so.3.7.1.0
>     |-- libtbb.so
>     |-- libtbb.so.2
>     |-- libtbb_preview.so
>     |-- libtbb_preview.so.2
>     |-- libtbbbind.so
>     |-- libtbbbind.so.2
>     |-- libtbbmalloc.so
>     |-- libtbbmalloc.so.2
>     |-- libtbbmalloc_proxy.so
>     |-- libtbbmalloc_proxy.so.2
>     |-- myriad_compile
>     |-- pcie-ma2x8x.mvcmd
>     |-- plugins.xml
>     `-- usb-ma2x8x.mvcmd
> ```
> 
> And due to the same I get an import error for the inference_engine module. Do I need to configure the build differently to enable this?
> 
> ```
> >>> from openvino.inference_engine import IECore, IENetwork
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> ModuleNotFoundError: No module named 'openvino.inference_engine'
> ```

Options are right.
Looks like python modules are missed in the wheel. @CoderHam could you please check the wheel build directory (build\inference-engine\ie_bridges\python\wheel), does it include site-packages\python\python3.8\openvino\inference_engine\ie_api.so? 

And could you please clean up the wheel build directory and repeat the build.

**Comment by CoderHam** [comment]
@jgespino is this issue fixed in 2021.3 or will it be fixed in 2021.4? I am still seeing the issue in 2021.2
